src/common/404/PageNotFound.jsx

```diff
@@ -1,25 +1,46 @@
+import { useState, useEffect } from "react";
 import { ReactComponent as Image404 } from "images/img-404.svg";
 import "./404.css";
 import Loader from "common/spinner/spinner";
+import { useNavigate } from "react-router-dom";
 
 const PageNotFound = ({ loading, msg, details, Image }) => {
+  const [timer, setTimer] = useState(5);
+  const navigate = useNavigate();
+
+  useEffect(() => {
+    const interval = setInterval(() => {
+      setTimer(timer - 1);
+    }, 1000);
+    return () => {
+      clearInterval(interval);
+      if (timer === 0) {
+        return navigate("/plays");
+      }
```

Comment:
@Angryman18 , Can we remove this logic from the cleanup function? because the cleanup function should not include the routing logic.

Reply:
i think as far as there is no error or warning we are good to go with it.